File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> st;
        int res = 0;

        // Hash all the array elements
        for (int val: nums)
            st.insert(val);

        // check each possible sequence from the start then update optimal length
        for (int val: nums) {
        
            // if current element is the starting element of a sequence
            if (st.find(val) != st.end() && st.find(val-1) == st.end()) {
            
                // Then check for next elements in the sequence
                int cur = val, cnt = 0;
                while (st.find(cur) != st.end()) {
                    
                    // Remove this number to avoid recomputation
                    st.erase(cur);
                    cur++;
                    cnt++;
                }

                // update  optimal length
                res = max(res, cnt);
            }
        }
        return res;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        unchecked_nums = set(nums)
        max_consecutive_length = 0
        
        for num in nums:
            if num - 1 not in unchecked_nums:
                current_length = 0
                current_num = num
                
                while current_num in unchecked_nums:
                    current_length += 1
                    unchecked_nums.remove(current_num)
                    current_num += 1

                max_consecutive_length = max(max_consecutive_length, current_length)
        
        return max_consecutive_length